packages/overlay/src/OverlayBase.ts

```diff
@@ -0,0 +1,908 @@
+            offset,
+            placement,
+            tipPadding,
+            trigger,
+            type: this.type,
+        });
+    }
+
+    protected async manageOpen(oldOpen: boolean): Promise<void> {
+        if (!this.isConnected && this.open) return;
+
+        if (!this.hasUpdated) {
+            await this.updateComplete;
+        }
+
+        if (this.open) {
+            overlayStack.add(this);
+            if (this.willPreventClose) {
+                document.addEventListener(
+                    'pointerup',
+                    () => {
+                        this.dialogEl.classList.toggle(
+                            'not-immediately-closable',
+                            false
+                        );
+                        this.willPreventClose = false;
+                    },
+                    { once: true }
+                );
+                this.dialogEl.classList.toggle(
+                    'not-immediately-closable',
+                    true
+                );
+            }
+        } else {
+            if (oldOpen) {
+                this.dispose();
+            }
+            overlayStack.remove(this);
+        }
+
+        if (this.usesDialog) {
+            this.manageDialogOpen();
+        } else {
+            this.managePopoverOpen();
+        }
+        if (!this.open) {
+            // If the focus remains inside of the overlay or
+            // a slotted descendent of the overlay you need to return
+            // focus back to the trigger.
+            const getAncestors = (): HTMLElement[] => {
+                const ancestors: HTMLElement[] = [];
+                // eslint-disable-next-line @spectrum-web-components/document-active-element
+                let currentNode = document.activeElement;
+                while (
+                    currentNode?.shadowRoot &&
+                    currentNode.shadowRoot.activeElement
+                ) {
+                    currentNode = currentNode.shadowRoot.activeElement;
+                }
+                while (currentNode) {
+                    const ancestor =
+                        currentNode.assignedSlot ||
+                        currentNode.parentElement ||
+                        (currentNode.getRootNode() as ShadowRoot)?.host;
+                    if (ancestor) {
+                        ancestors.push(ancestor as HTMLElement);
+                    }
+                    currentNode = ancestor;
+                }
+                return ancestors;
+            };
+            if (
+                (this.triggerElement as HTMLElement)?.focus &&
+                (this.contains(
+                    (this.getRootNode() as Document).activeElement
+                ) ||
+                    !!getAncestors().find((el) => el === this))
+            ) {
+                (this.triggerElement as HTMLElement).focus();
+            }
+        }
+    }
+
+    protected unbindEvents(triggerElement: HTMLElement): void {
```

Comment:
There's a lot of bookkeeping here... if there are discrete, logically-related groups of listeners, it might make sense to cancel them via one or more [AbortControllers](https://developer.mozilla.org/en-US/docs/Web/API/AbortController).

Reply:
And here I just change the signature of these methods to be reciprocal for you 😉

Reply:
I did see that!